### ISSUE TITLE
MacOS wrong environment path - Crash without any solution

### ISSUE THREAD

**Issue by brodienguyen** [jupyterlab/jupyterlab-desktop#599]
## Description

I think I have entered the wrong path for the custom env variable. However, since the 2nd time, the app will crash immediately doesn't matter what I tried (uninstall, download new one -> install, shutdown, restart). Refer to the following video, the app icon just flash for 1 second then terminate it.

![IMTBN 2023-02-28 at 23 43 08](https://user-images.githubusercontent.com/32996052/221857642-8c9c3821-a426-4cba-a7c0-3dad0f2fab93.gif)

## Reproduce

1. Download the latest MacOS version of Jupyter Lab
2. Open it from Downloads folder
3. Paste the custom path (I think I have entered the wrong path for the custom environment but not sure how to change it)
4. Next time, I have open the app it will crash immediately. I have tried to delete it then download the new one but it doesn't help.

## Expected behavior

- The Jupyter app should be opened properly

## Context

- Operating System and version: 13.2.1 
- JupyterLab-Desktop version: v3.6.1-2
- Installer: dmg

<details><summary>Troubleshoot Output</summary>
<pre>
[notice] A new release of pip available: 22.3.1 -> 23.0.1
[notice] To update, run: pip install --upgrade pip
$PATH:
	/Users/brodienguyen/.pyenv/versions/3.11.1/bin
	/usr/local/Cellar/pyenv/2.3.13/libexec
	/usr/local/Cellar/pyenv/2.3.13/plugins/python-build/bin
	/Users/brodienguyen/.pyenv/shims
	/Users/brodienguyen/.nvm/versions/node/v19.6.0/bin
	/Users/brodienguyen/.rbenv/shims
	/usr/local/bin
	/System/Cryptexes/App/usr/bin
	/usr/bin
	/bin
	/usr/sbin
	/sbin
	/Library/Apple/usr/bin
	/Applications/Postgres.app/Contents/Versions/latest/bin
	/Users/brodienguyen/.fig/bin
	/Users/brodienguyen/.local/bin
	/Users/brodienguyen/fvm/versions/3.7.3/bin

sys.path:
	/Users/brodienguyen/.pyenv/versions/3.11.1/bin
	/Users/brodienguyen/.pyenv/versions/3.11.1/lib/python311.zip
	/Users/brodienguyen/.pyenv/versions/3.11.1/lib/python3.11
	/Users/brodienguyen/.pyenv/versions/3.11.1/lib/python3.11/lib-dynload
	/Users/brodienguyen/.pyenv/versions/3.11.1/lib/python3.11/site-packages

sys.executable:
	/Users/brodienguyen/.pyenv/versions/3.11.1/bin/python3.11

sys.version:
	3.11.1 (main, Feb 28 2023, 23:19:39) [Clang 14.0.0 (clang-1400.0.29.202)]

platform.platform():
	macOS-13.2.1-x86_64-i386-64bit

which -a jupyter:
	/Users/brodienguyen/.pyenv/versions/3.11.1/bin/jupyter
	/Users/brodienguyen/.pyenv/shims/jupyter

pip list:
	Package                  Version
	------------------------ ---------
	aiofiles                 22.1.0
	aiosqlite                0.18.0
	anyio                    3.6.2
	appnope                  0.1.3
	argon2-cffi              21.3.0
	argon2-cffi-bindings     21.2.0
	arrow                    1.2.3
	asttokens                2.2.1
	attrs                    22.2.0
	Babel                    2.12.0
	backcall                 0.2.0
	beautifulsoup4           4.11.2
	bleach                   6.0.0
	certifi                  2022.12.7
	cffi                     1.15.1
	charset-normalizer       3.0.1
	comm                     0.1.2
	debugpy                  1.6.6
	decorator                5.1.1
	defusedxml               0.7.1
	executing                1.2.0
	fastjsonschema           2.16.3
	fqdn                     1.5.1
	idna                     3.4
	ipykernel                6.21.2
	ipython                  8.10.0
	ipython-genutils         0.2.0
	isoduration              20.11.0
	jedi                     0.18.2
	Jinja2                   3.1.2
	json5                    0.9.11
	jsonpointer              2.3
	jsonschema               4.17.3
	jupyter_client           8.0.3
	jupyter_core             5.2.0
	jupyter-events           0.6.3
	jupyter_server           2.3.0
	jupyter_server_fileid    0.8.0
	jupyter_server_terminals 0.4.4
	jupyter_server_ydoc      0.6.1
	jupyter-ydoc             0.2.2
	jupyterlab               3.6.1
	jupyterlab-pygments      0.2.2
	jupyterlab_server        2.19.0
	MarkupSafe               2.1.2
	matplotlib-inline        0.1.6
	mistune                  2.0.5
	nbclassic                0.5.2
	nbclient                 0.7.2
	nbconvert                7.2.9
	nbformat                 5.7.3
	nest-asyncio             1.5.6
	notebook                 6.5.2
	notebook_shim            0.2.2
	packaging                23.0
	pandocfilters            1.5.0
	parso                    0.8.3
	pexpect                  4.8.0
	pickleshare              0.7.5
	pip                      22.3.1
	platformdirs             3.0.0
	prometheus-client        0.16.0
	prompt-toolkit           3.0.38
	psutil                   5.9.4
	ptyprocess               0.7.0
	pure-eval                0.2.2
	pycparser                2.21
	Pygments                 2.14.0
	pyrsistent               0.19.3
	python-dateutil          2.8.2
	python-json-logger       2.0.7
	PyYAML                   6.0
	pyzmq                    25.0.0
	requests                 2.28.2
	rfc3339-validator        0.1.4
	rfc3986-validator        0.1.1
	Send2Trash               1.8.0
	setuptools               65.5.0
	six                      1.16.0
	sniffio                  1.3.0
	soupsieve                2.4
	stack-data               0.6.2
	terminado                0.17.1
	tinycss2                 1.2.1
	tornado                  6.2
	traitlets                5.9.0
	uri-template             1.2.0
	urllib3                  1.26.14
	wcwidth                  0.2.6
	webcolors                1.12
	webencodings             0.5.1
	websocket-client         1.5.1
	y-py                     0.5.9
	ypy-websocket            0.8.2

</pre>
</details>

<details><summary>Browser Output (required for interface issues after loading)</summary>

<pre>
Paste the output from the DevTools JavaScript console here, if applicable.

</pre>
</details>

<details><summary>Logs (required for issues with loading, server and startup)</summary>

<pre>
[2023-02-28 23:25:59.448] [error] Error: EROFS: read-only file system, chmod '/usr/local/bin/jlab'
    at Object.chmodSync (node:fs:1880:3)
    at setupJLabCommand (/private/var/folders/h6/lnht_bz503bdl_qk_g68y10w0000gn/T/AppTranslocation/59ADCB30-6A82-4D23-B82B-CFBF3952B0A5/d/JupyterLab.app/Contents/Resources/app.asar/build/out/main/main.js:152:16)
    at /private/var/folders/h6/lnht_bz503bdl_qk_g68y10w0000gn/T/AppTranslocation/59ADCB30-6A82-4D23-B82B-CFBF3952B0A5/d/JupyterLab.app/Contents/Resources/app.asar/build/out/main/main.js:199:9
[2023-02-28 23:25:59.467] [error] SyntaxError: Unexpected token 'E', "EXEC:ERROR" is not valid JSON
    at JSON.parse (<anonymous>)
    at Registry.getEnvironmentInfoSync (/private/var/folders/h6/lnht_bz503bdl_qk_g68y10w0000gn/T/AppTranslocation/59ADCB30-6A82-4D23-B82B-CFBF3952B0A5/d/JupyterLab.app/Contents/Resources/app.asar/build/out/main/registry.js:305:30)
    at Registry._resolveEnvironmentSync (/private/var/folders/h6/lnht_bz503bdl_qk_g68y10w0000gn/T/AppTranslocation/59ADCB30-6A82-4D23-B82B-CFBF3952B0A5/d/JupyterLab.app/Contents/Resources/app.asar/build/out/main/registry.js:168:26)
    at new Registry (/private/var/folders/h6/lnht_bz503bdl_qk_g68y10w0000gn/T/AppTranslocation/59ADCB30-6A82-4D23-B82B-CFBF3952B0A5/d/JupyterLab.app/Contents/Resources/app.asar/build/out/main/registry.js:70:33)
    at new JupyterApplication (/private/var/folders/h6/lnht_bz503bdl_qk_g68y10w0000gn/T/AppTranslocation/59ADCB30-6A82-4D23-B82B-CFBF3952B0A5/d/JupyterLab.app/Contents/Resources/app.asar/build/out/main/app.js:175:26)
    at /private/var/folders/h6/lnht_bz503bdl_qk_g68y10w0000gn/T/AppTranslocation/59ADCB30-6A82-4D23-B82B-CFBF3952B0A5/d/JupyterLab.app/Contents/Resources/app.asar/build/out/main/main.js:202:22

</pre>
</details>

**Comment by krassowski** [comment]
Is there a `.jupyter` folder in the directory you start the app from that you could try removing?

**Comment by mbektas** [comment]
@brodienguyen you can delete the `User settings` and `Application data` files as well in [these locations](https://github.com/jupyterlab/jupyterlab-desktop/blob/master/user-guide.md#configuration-and-data-files).

**Comment by JasonWeill** [comment]
Looks like JupyterLab is running inside a special sandbox directory, as done by recent macOS versions. The feature, [app translocation](https://eclecticlight.co/2022/09/06/dont-run-that-app-where-it-landed-how-translocation-can-cause-crashes/), has caused crashes in other apps.

**Comment by brodienguyen** [comment]
> @brodienguyen you can delete the `User settings` and `Application data` files as well in [these locations](https://github.com/jupyterlab/jupyterlab-desktop/blob/master/user-guide.md#configuration-and-data-files).

I am unable to find this since the app has been move to Applications before. Basically I just download the dmg -> the file landed on my Downloads -> I double click to open -> set the custom path to pyenv (but I may put a wrong path) -> The app restart -> The crash happened in a loop.

**Comment by brodienguyen** [comment]
> Is there a `.jupyter` folder in the directory you start the app from that you could try removing?

I can't locate this file unfortunately

**Comment by brodienguyen** [comment]
> Looks like JupyterLab is running inside a special sandbox directory, as done by recent macOS versions. The feature, [app translocation](https://eclecticlight.co/2022/09/06/dont-run-that-app-where-it-landed-how-translocation-can-cause-crashes/), has caused crashes in other apps.

Is there any potential solution on the upcoming updates please? Thanks heaps for following it up

**Comment by JasonWeill** [comment]
> I am unable to find this since the app has been move to Applications before. Basically I just download the dmg -> the file landed on my Downloads -> I double click to open -> set the custom path to pyenv (but I may put a wrong path) -> The app restart -> The crash happened in a loop.

@brodienguyen Are you running JupyterLab Desktop directly from the disk image? If so, that may be where the errors are coming from. You should be copying the application from the disk image to your Applications folder, then running JupyterLab Desktop from Applications.

**Comment by brodienguyen** [comment]
> > I am unable to find this since the app has been move to Applications before. Basically I just download the dmg -> the file landed on my Downloads -> I double click to open -> set the custom path to pyenv (but I may put a wrong path) -> The app restart -> The crash happened in a loop.
> 
> @brodienguyen Are you running JupyterLab Desktop directly from the disk image? If so, that may be where the errors are coming from. You should be copying the application from the disk image to your Applications folder, then running JupyterLab Desktop from Applications.

You are absolutely right. I did run it directly from the Downloads folder. So could you advise me what I should do now to fix this issue please?

**Comment by JasonWeill** [comment]
@brodienguyen Try opening (mounting) the disk image, copying the JupyterLab Desktop icon to the Applications folder (you can drag it to the "Applications" alias in the disk image), then ejecting the disk image. Then, go to your Applications folder or [Launchpad](https://support.apple.com/guide/mac-help/open-apps-with-launchpad-mh35840/mac), and click the JupyterLab Desktop icon.

**Comment by JasonWeill** [comment]
@brodienguyen When you move JupyterLab's application to the Applications folder, then run it from that folder, do you still see this behavior?

**Comment by JasonWeill** [comment]
@brodienguyen I haven't seen any response on this issue since March 9, so I'm closing it. Please reopen or open a new issue if you still see this problem. Thanks!

**Comment by brodienguyen** [comment]
@JasonWeill Sorry that somehow I lost the notification on this issue. I have retried and doesn't seem to work. I just want to do some experiment with Python so I ended up to use jupyter notebook instead for now. I will come back if I intend to work on some project require Python in the near future. Thank you for your time and again, apologize for not following up this conversation.

Have a good one from Australia 🇦🇺
